configure

```diff
@@ -0,0 +1,85 @@
+#! /bin/sh
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License.  You may obtain a copy
+# of the License at:
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# This is a shell script and not a Go program because we have to run
+# bazel-gopath, and the bazel-gopath tool invokes Bazel on its own.  So, if
+# this were a Go program and we wanted to tell users to invoke it with "bazel
+# run" as we do for all other tools, bazel-gopath would get stuck because Bazel
+# can only run one command at a time.  It's easier for now to just keep this as
+# a shell-script so that it's impossible to run it from within Bazel.
+#
+# TODO(jmmv): Convert to a Go program once bazel-gopath is fixed to not need
+# to run Bazel.  (E.g. it could be made to read query outputs from a dependency
+# and then have a Bazel genquery rule to supply such data as a dependency.)
+
+set -e
```

Comment:
set -euo pipefail?

Reply:
(Just bumping this because GitHub says the previous review was dismissed)

Is it not safe to use set -euo pipefail for the configure script?

Reply:
Wut? I'm sure I typed a reply here!

`set -u`: This would be nice... but the ancient bash (which is `/bin/sh`) shipped on macOS has bugs that cause it to complain on references to `"${@}"` when there are no arguments — and references to empty `"${@}"` should be valid. This makes the call to main, and the argument processing, to fail when this is enabled.

`set -o pipefail`: This is a bash-only feature and I don't want to rely on bash here.